### PR TITLE
EOS-13993: NFS ADDB: ADDB for all file operations

### PR DESCRIPTION
**Pull request by vimalraghwani20**
**Problem Statement**
EOS-13993: NFS ADDB: ADDB for all file operations

**Problem Description**
Add ADDB traces for all file operations

**Solution Overview**
Added ADDB traces for following file operations

- open
- close
- status
- reopen
- commit

**Companion PRs**
https://github.com/Seagate/cortx-fs/pull/50
https://github.com/Seagate/cortx-utils/pull/68

### CODE REVIEW

File: src/FSAL/FSAL_CORTXFS/handle.c [hunk 1]
```diff
@@ -1926,6 +2025,8 @@ static fsal_status_t kvsfs_share_try_new_state(struct kvsfs_fsal_obj_handle *obj
 {
 	fsal_status_t status;
 
+	perfc_trace_inii(PFT_KVSFS_SHARE_TRY_NEW_STATE, PEM_KVSFS_TO_NFS);
```
Comment (pratyush-seagate) [inline src/FSAL/FSAL_CORTXFS/handle.c:2028]: KVSFS tags should not be used from handle.c file.

File: src/FSAL/FSAL_CORTXFS/handle.c [hunk 2]
```diff
@@ -1977,10 +2079,14 @@ static fsal_status_t cfs_file_open(struct kvsfs_file_state *state,
 				   fsal_openflags_t openflags,
 				   struct kvsfs_fsal_obj_handle *obj)
 {
+	fsal_status_t result;
+	perfc_trace_inii(PFT_CFS_OPEN, PEM_CFS_TO_NFS);
```
Comment (pratyush-seagate) [inline src/FSAL/FSAL_CORTXFS/handle.c:2083]: CFS tags should not be used from handle.c file

File: src/FSAL/FSAL_CORTXFS/handle.c [hunk 3]
```diff
@@ -1990,9 +2096,13 @@ static fsal_status_t cfs_file_open(struct kvsfs_file_state *state,
 static fsal_status_t cfs_file_close(struct kvsfs_file_state *state,
 				    struct kvsfs_fsal_obj_handle *obj)
 {
+	fsal_status_t result;
+	perfc_trace_inii(PFT_CFS_CLOSE, PEM_CFS_TO_NFS);
```
Comment (pratyush-seagate) [inline src/FSAL/FSAL_CORTXFS/handle.c:2100]: CFS tags should not be used from this file

File: src/FSAL/FSAL_CORTXFS/handle.c [hunk 4]
```diff
@@ -2005,6 +2115,7 @@ static fsal_status_t kvsfs_file_state_open(struct kvsfs_file_state *state,
 {
 	fsal_status_t status;
 
+	perfc_trace_inii(PFT_KVSFS_FILE_STATE_OPEN, PEM_KVSFS_TO_NFS);
```
Comment (pratyush-seagate) [inline src/FSAL/FSAL_CORTXFS/handle.c:2118]: KVSFS tags!

File: src/FSAL/FSAL_CORTXFS/handle.c [hunk 5]
```diff
@@ -2350,6 +2462,7 @@ static fsal_status_t kvsfs_open2_by_handle(struct fsal_obj_handle *obj_hdl,
 	struct kvsfs_file_state *fd;
 	struct stat stat;
 
+	perfc_trace_inii(PFT_KVSFS_OPEN2_BY_HANDLE, PEM_KVSFS_TO_NFS);
```
Comment (pratyush-seagate) [inline src/FSAL/FSAL_CORTXFS/handle.c:2465]: KVSFS tags!

File: src/FSAL/FSAL_CORTXFS/handle.c [hunk 6]
```diff
@@ -2394,6 +2508,7 @@ static fsal_status_t kvsfs_open2_by_name(struct fsal_obj_handle *parent_obj_hdl,
 	fsal_status_t result;
 	struct fsal_obj_handle *obj = NULL;
 
+	perfc_trace_inii(PFT_KVSFS_OPEN2_BY_NAME, PEM_KVSFS_TO_NFS);
```
Comment (pratyush-seagate) [inline src/FSAL/FSAL_CORTXFS/handle.c:2511]: KVSFS tags!

File: src/FSAL/FSAL_CORTXFS/handle.c [hunk 7]
```diff
@@ -2497,6 +2613,7 @@ kvsfs_create_unchecked(struct fsal_obj_handle *parent_obj_hdl, const char *name,
 
 	T_ENTER(">>> (parent=%p, state=%p, name=%s, attrs_in=%p, attrs_out=%p)",
 		parent_obj_hdl, state, name, attrs_in, attrs_out);
+	perfc_trace_inii(PFT_KVSFS_CREATE_UNCHECKED, PEM_KVSFS_TO_NFS);
```
Comment (pratyush-seagate) [inline src/FSAL/FSAL_CORTXFS/handle.c:2616]: KVSFS tags!

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 1 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c:2028]: which is convenient function tag to use from handle.c?

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 2 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c:2083]: which is convenient function tag to use from handle.c?

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 3 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c:2100]: which is convenient function tag to use from handle.c?

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 4 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c:2118]: which is convenient function tag to use from handle.c?

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 1 above)
Comment (pratyush-seagate) [reply src/FSAL/FSAL_CORTXFS/handle.c:2028]: Pls remove KVSFS sub string

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 2 above)
Comment (pratyush-seagate) [reply src/FSAL/FSAL_CORTXFS/handle.c:2083]: Pls remove CFS sub string, i.e. PFT_FILE_OPEN

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 3 above)
Comment (pratyush-seagate) [reply src/FSAL/FSAL_CORTXFS/handle.c:2100]: Same as previous

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 4 above)
Comment (pratyush-seagate) [reply src/FSAL/FSAL_CORTXFS/handle.c:2118]: Same as previous

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 7 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c]: fixed

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 6 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c]: fixed

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 5 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c]: fixed

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 4 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c]: fixed

File: src/FSAL/FSAL_CORTXFS/handle.c (hunk 1 above)
Comment (vimalraghwani20) [reply src/FSAL/FSAL_CORTXFS/handle.c]: fixed

**Review by pratyush-seagate — APPROVED** [review]
LGTM, +1

**Review by AtitaShirwaikar — APPROVED** [review]
+1

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**